tests/e2e/pages/vscode.page.ts

```diff
@@ -1,43 +1,198 @@
+   * Closes the VSCode instance.
+   */
+  public async closeVSCode(): Promise<void> {
+    try {
+      if (this.app) {
+        await this.app.close();
+        console.log('VSCode closed successfully.');
+      }
+    } catch (error) {
+      console.error('Error closing VSCode:', error);
+    }
+  }
+
+  /**
+   * Waits for the Konveyor extension to complete initialization by watching for
+   * the __EXTENSION_INITIALIZED__ info message signal.
    */
-  public abstract writeOrUpdateVSCodeSettings(settings: Record<string, any>): Promise<void>;
-  protected abstract selectCustomRules(customRulesPath: string): Promise<void>;
-  public abstract closeVSCode(): Promise<void>;
-  public abstract pasteContent(content: string): Promise<void>;
-  public abstract getWindow(): Page;
+  public async waitForExtensionInitialization(): Promise<void> {
+    try {
+      console.log('Waiting for Konveyor extension initialization...');
+
+      const javaReadySelector = this.getWindow().getByRole('button', { name: 'Java: Ready' });
+
+      await javaReadySelector.waitFor({ timeout: 120000 });
+      // Sometimes the java ready status is displayed for a few seconds before starting to load again
+      // This checks that the state is kept for a few seconds before continuing
+      await this.waitDefault();
+      await javaReadySelector.waitFor({ timeout: 1200000 });
+
+      // Trigger extension activation by opening the analysis view
+      // This was working before - the extension activates and opens the view
+      await this.executeQuickCommand(`${COMMAND_CATEGORY}: Open Analysis View`);
+
+      // Now wait for the initialization signal message to appear
+      // This message is shown by the extension when __TEST_EXTENSION_END_TO_END__ env var is set
+      const initializationMessage = this.window
+        .getByRole('alert')
+        .getByText('__EXTENSION_INITIALIZED__');
+      await expect(initializationMessage).toBeVisible({ timeout: 300000 }); // 5 minute timeout for asset downloads
+
+      // Dismiss the message
+      await this.window.keyboard.press('Escape');
+      await this.window.waitForTimeout(2000); // Give VSCode a chance to process the message
+      console.log('Konveyor extension initialized successfully');
+    } catch (error) {
+      console.error('Failed to wait for extension initialization:', error);
+      throw error;
+    }
+  }
 
   public async executeQuickCommand(command: string) {
     await this.waitDefault();
     const modifier = getOSInfo() === 'macOS' ? 'Meta' : 'Control';
     await this.window.keyboard.press(`${modifier}+Shift+P`, { delay: 500 });
     const input = this.window.getByPlaceholder('Type the name of a command to run.');
     await input.fill(`>${command}`);
-    await expect(
-      this.window.locator(`a.label-name span.highlight >> text="${command}"`)
-    ).toBeVisible();
-
+    await expect(this.window.locator('a').filter({ hasText: command })).toBeVisible({
```

Comment:
`hasText` searches for substring as well so this may not work when there are 2 similar commands, like:

```
Konveyor: Open the default the GenAI model provider configuration file and backup the current file
Konveyor: Open the GenAI model provider configuration file
```